apps/src/music/views/KeyHandler.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {useContext, useEffect} from 'react';
+import {useDispatch} from 'react-redux';
+import {Triggers} from '../constants';
+import {AnalyticsContext} from '../context';
+import {
+  advanceInstructionsPosition,
+  toggleBeatPad,
+  toggleInstructions,
+  toggleTimelinePosition
+} from '../redux/musicRedux';
+
+interface KeyHandlerProps {
+  togglePlaying: () => void;
+  playTrigger: (triggerId: string) => void;
+}
+
+/**
+ * Utility component for handling key presses in Music Lab that map to
+ * specific UI actions.
+ */
+const KeyHandler: React.FunctionComponent<KeyHandlerProps> = ({
+  togglePlaying,
+  playTrigger
+}) => {
+  const analyticsReporter = useContext(AnalyticsContext);
+  const dispatch = useDispatch();
+
+  const handleKeyUp = (event: KeyboardEvent) => {
+    // Don't handle a keyboard shortcut if the active element is an
+    // input field, since the user is probably trying to type something.
+    if (
+      document.activeElement &&
+      document.activeElement.tagName.toLowerCase() === 'input'
+    ) {
+      return;
+    }
+
+    // When assigning new keyboard shortcuts, be aware that the following
+    // keys are used for Blockly keyboard navigation: A, D, I, S, T, W, X
+    // https://developers.google.com/blockly/guides/configure/web/keyboard-nav
+    // Also avoid C and V that may be used in copy/paste shortcuts.
```

Comment:
Presumably there's a way to see if the C & V are being used as part of a key-combo rather than standalone?

Reply:
Good point - I actually didn't know we were hitting this issue until the school visit yesterday, so it may be something unique to Edge/Windows. I can play around with it a bit more though to check.